packages/advi/advi.2.0.0/opam

```diff
@@ -0,0 +1,51 @@
+opam-version: "2.0"
+synopsis: "Active DVI Dune package!"
+description: """
+
+Active-DVI is a DVI previewer and presenter written in Objective Caml
+(http://caml.inria.fr/ocaml/) with some eye candy effects for
+presentation, support for interactive demonstrations, and embedding of
+arbitrary applications within the presentation.
+
+Active-DVI is very well suited for use in combination with WhizzyTeX
+(http://cristal.inria.fr/whizzytex/)
+"""
+maintainer: ["Didier Rémy <Didier.Remy@inria.fr>"]
+authors: [
+  "Alexandre Miquel"
+  "Jun Furuse"
+  "Didier Rémy"
+  "Pierre Weis"
+  "Xavier Leroy"
+  "Roberto Di Cosmo"
+  "Didier Le Botlan"
+  "Alan Schmitt"
+]
+license: "<name>"
+homepage: "http://advi.inria.fr"
+doc: "https://gitlab.inria.fr/remy/advi"
+bug-reports: "Didier.Remy@inria.fr"
+depends: [
+  "ocaml-base-compiler" {>= "4.11.1"}
+  "dune" {>= "2.5"}
+  "graphics" {>= "5.1.1"}
+  "camlimages" {>= "5.0.4"}
+]
```

Comment:
```suggestion
  "conf-texlive"
]
```

needed to pull in kpsewhich

Reply:
Thanks. I did not know about this package.   

However, this package only checks that the command pdflatex exists, while I instead need to check that the command "kpsewhich" exists.  The former may imply the later, but I am not entirely sure: there may be installations of latex that do not install pdflatex (which did not exist when advi was first written), athough I would be surprised if it were the case today.  

Conversely, I am not sure that the presence of pdflatex ensures that of kpsewhich, although again I would be surprised if it were not the case. 

Another question is why isn't there a depext formular for macports and just one for homebrew? 

Actually, the defaut detection of os-family by opam seems biaised towards homebrew: I have a full installation with macports and an empty installation of homebrew (with zero homebrew package installed) and I had to explicitly tell opam that my os-distribution family is macports, otherwise it could not compile conf-libssl. 

> Many thanks for the contribution, @diremy! For this to pass CI (and for users with the `opam depext` plugin), it would be useful to have the OS dependencies recorded. 

> The build currently fails with:
> 
> ```
> #=== ERROR while compiling advi.2.0.0 =========================================#
> # context              2.0.8 | linux/x86_64 | ocaml-base-compiler.4.11.2 | pinned(https://github.com/diremy/advi/archive/advi-2.0.0.tar.gz)
> # path                 ~/.opam/4.11/.opam-switch/build/advi.2.0.0
> # command              ~/.opam/opam-init/hooks/sandbox.sh build make
> # exit-code            2
> # env-file             ~/.opam/log/advi-15-ecfa18.env
> # output-file          ~/.opam/log/advi-15-ecfa18.out
> ### output ###
> # dune build --root=. src/Makefile.config
> #         bash src/configure.sh (exit 1)
> # (cd _build/default/src && /bin/bash discover.sh configure.sh)
> # No kpsewhich command found, check your setup!
> # make: *** [Makefile:28: src/Makefile.config] Error 1
> ```
> 
> I left a suggestion in the PR to add the conf-texlive package which already has the external dependencies for this. If you commit that suggestion to your branch, the CI will retest.

How should I proceed? Should I fix this on advi git and then resubmit another independent opam-publish pull request or is there another better way? 

Didier

Reply:
You can modify the metadata in the opam package of this PR without having to retag and resubmit. Opam uses the opam-repository information for the installation, not the file in the repository (unless you pin that).

In general any recent enough texlive installation should come with `kpsewhich`, so I think the suggestion is the best bet. I don’t know if there is a separate package that you can specify for it, but it seems unnecessarily complicated and can always be improved in the future if needed.

as per brew vs macports, usually people use one or the other so this has never come up. I think it may be worth opening an issue and see if we can find a better default. I think brew becomes preferred mostly because appears earlier alphabetically.

Reply:
@diremy I have committed the suggested change, let’s see if it is enough to unblock the tests. If so please make the change also in your repository (no need to re tag or republish)

Reply:
The archlinux failure is surprising since from documentation it seems that texlive should ship the command: https://wiki.archlinux.org/index.php/TeX_Live